File: Programmers/Lv0/중복된문자제거.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string solution(string my_string) {
    std::vector<char> my_vector;
    for (char ch : my_string) {
        if (find(my_vector.begin(), my_vector.end(), ch) == my_vector.end()) {
            my_vector.push_back(ch);
        }
    }
    string answer(my_vector.begin(), my_vector.end());
    return answer;
}

File: Programmers/Lv0/중복된문자제거.py
# set은 자동정렬되므로 정렬이 되지 않는 딕셔너리 구조를 이용
from collections import OrderedDict

def solution(my_string):
    my_set = list(OrderedDict.fromkeys(my_string))
    answer = ''.join(my_set)
    return answer